worker/src/RTC/RtpStreamRecv.cpp

```diff
@@ -645,9 +645,11 @@ namespace RTC
 		if (this->params.clockRate == 0u)
 			return;
 
-		auto transit =
-		  static_cast<int>(DepLibUV::GetTimeMs() - (rtpTimestamp * 1000 / this->params.clockRate));
-		int d = transit - this->transit;
+		// NOTE: Based on https://github.com/versatica/mediasoup/issues/1018.
+		double transitFactor =
+		  static_cast<double>(rtpTimestamp) * 1000.0 / static_cast<double>(this->params.clockRate);
+		auto transit = static_cast<int>(DepLibUV::GetTimeMs() - transitFactor);
```

Comment:
@nazar-pc would you mind tell me how this would be done in Rust? This is a good change to take advantage of this eventual hatred that I feel towards C++.

Reply:
Rust refuses to compile if types don't match exactly, which is one of the most confusing parts of C++ for me.

For overflows it will panic in debug build and have some deterministic non-panicking behavior in release build (I believe it will just wrap around) for performance reasons, but you can make it panic in release too if you want (or use checked math to handle it explicitly, see https://doc.rust-lang.org/stable/std/primitive.u32.html#method.checked_add for instance), that is just not the default behavior.

There are methods for checked, overflowing, wrapping, saturating math, so if you want to have exact control over what is happening to numbers you have a decent toolbox. We use those heavily in blockchain code where accidental invisible overflow can have catastrophic consequences.

Some overflow cases are detected by compiler and generate warnings, but that is for something trivial like `1u32 >> 32`.

So while I wouldn't necessarily prevent something like this, it would likely make it easier to discover or handle properly. You can just build an app with panicking overflows and fuzz it with inputs, uncovering various edge-cases.

Reply:
With the templates and inline operators, one can make his own type-safe arithmetics in C++ if he wishes. C++ is intended, among other things, to adapt easily existing C code (and familiar programming style), hence these implicit type conversions.